x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,718 @@
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. Start the node where you updated the keystore.
+
+. [[verify-keystore-newca,step 6]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that {es} loaded the new keystore.
+
+. If you're only updating certificates for the transport layer (and not the HTTP
+layer), then complete <<start-rolling-restart-newca>> through
+<<verify-keystore-newca>> one node at a time until you've updated all keystores
+in your cluster. You can then complete the remaining steps for a
+<<restart-cluster-rolling,rolling restart>>.
++
+Otherwise, do not complete a rolling restart and proceed to the steps for
+generating a new certificate for the HTTP layer.
+
+. (Optional) After rotating keystores on each node in your cluster,
+<<check-ca-truststore,list the certificates in your truststore>> and then remove
+the old CA certificate.
++
+If you previously used the `elasticsearch-certutil` tool to generate your
+keystore, the alias of the old CA defaults to `ca` and the type of entry is
+`PrivateKeyEntry`.
++
+[source,shell]
+----
+keytool -delete -noprompt -alias ca  -keystore config/elastic-stack-ca.p12 \
+-storepass <password>
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `-alias`:: Name of the keystore alias for the old CA certificate that you want to remove from your
+   truststore.
+====
+
+[discrete]
+[[transport-layer-newca-whatsnext]]
+==== What's next?
+Well done! You've updated the keystore for the transport layer. You can also
+<<node-certs-different-http,update the keystore for the HTTP layer>> if
+necessary. If you're not updating the keystore for the HTTP layer, then you're
+all set. 
+
+[[node-certs-different-http]]
+==== Generate a new certificate for the HTTP layer
+You can generate certificates for the HTTP layer using your new CA certificate
+and private key. Other components such as {kib} or any of the Elastic language
+clients verify this certificate when they connect to {es}.
+
+NOTE: If your organization has its own CA, you'll need to 
+<<certutil-csr,generate Certificate Signing Requests (CSRs)>>. CSRs contain
+information that your CA uses to generate and sign a security certificate
+instead of using self-signed certificates that the `elasticsearch-certutil` tool
+generates.
+
+Before generating new certificates for the HTTP layer, you need to go to all the clients that connect to {es} (such as {kib}, {beats}, {ls}, and any language
+clients) and configure them to also trust the new CA (`ca.crt`) that you
+generated.
+
+. For *every* client that connects to {es}, import the new `ca.crt` certificate
```

Comment:
let's rewrite this. This is an action that must be taken on the client side, not on elasticserach. The new `ca.crt` is one, so it doesn't make sense to import it again and again, once for every client there is , into the same keystore. We're referencing `http.p12` here but we haven't created it yet.  FWIW the http.p12 keystore will eventually contain the new ca.crt certiifcate as part of the certificate chain in the private key entry, certutil takes care of that.

It's the client that needs to trust the certificate, so it;s the client that needs to be configured to do so. The way you do this in every client is different, so I don't know if we can enumerate all these here.

Reply:
Ah, understood! So we would remove the next two steps (importing `ca.crt` and checking that it was imported) and instead say:
> For *every* client that connects to {es}, configure the client to trust the `ca.crt` certificate. This process is different for each client, so refer to your client's documentation for trusting certificates.

Reply:
++